src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -89,6 +96,15 @@ public ModelMetadata(StreamInput in) throws IOException {
         } else {
             this.vectorDataType = VectorDataType.DEFAULT;
         }
+
+        if (IndexUtil.isVersionOnOrAfterMinRequiredVersion(in.getVersion(), KNNConstants.MINIMAL_MODE_AND_COMPRESSION_FEATURE)) {
+            this.mode = Mode.fromString(in.readOptionalString());
+            this.compressionLevel = CompressionLevel.fromString(in.readOptionalString());
+        } else {
+            this.mode = Mode.NOT_CONFIGURED;
+            this.compressionLevel = CompressionLevel.NOT_CONFIGURED;
```

Comment:
nit: could have defaulted to `in_memory` and `1x` here. They are still valid values for before 2.17

Reply:
I prefer NOT_CONFIGURED just because I want to differentiate when someone actually supplies as config "1x" or "in_memory" and handle that case a bit differently.

Reply:
> handle that case a bit differently

Would appreciate more context on this part. I was of the opinion it will execute the same code path as the current one